From a05c66dc913c58394ed2af4695b01fb3e00a1b90 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 30 Mar 2021 23:19:03 +0300
Subject: [PATCH] [SW-28697] lkd: move noret outside noret_legacy()

The implementation of the LKD module "no return" commands is about to
stay the same for legacy and the new protocol.
As a preparation for the new LKD protocol addition and "no return"
functionality implementation reuse, move the "no return" commands
implementation to separate functions.

Change-Id: Ie1e4cbfaa9ff936e845c4e451ca96ff83a57dc26
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/preboot/module_lkd.c b/preboot/module_lkd.c
index daf36b6..b9ca6b1 100644
--- a/preboot/module_lkd.c
+++ b/preboot/module_lkd.c
@@ -148,24 +148,38 @@
 /* For devices that do not implement RST_DEV command, mimic WFE */
 __weak void hl_dev_reset(void) { wfe(); }
 
+static void module_lkd_run_noret_exec(void)
+{
+	preboot_security_cleanup_before_run();
+	hl_ts_shutdown();
+	/* We do not know if the FW we are loading supports hard rst */
+	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_FW_HARD_RST_EN, false);
+	fit_run_image();
+}
+
+static void module_lkd_run_noret_reset(void)
+{
+	cpu_boot_sts_set(CPU_BOOT_STATUS_IN_WFE);
+	hl_dev_reset();	/* should not return */
+}
+
+static void module_lkd_run_noret_wfe(void)
+{
+	cpu_boot_sts_set(CPU_BOOT_STATUS_IN_WFE);
+	wfe();
+}
+
 static void module_lkd_run_noret_legacy(void)
 {
 	switch (sts.cmd.val) {
 	case KMD_MSG_GOTO_WFE:
-		cpu_boot_sts_set(CPU_BOOT_STATUS_IN_WFE);
-		wfe();
+		module_lkd_run_noret_wfe();
 		break;
 	case KMD_MSG_RST_DEV:
-		cpu_boot_sts_set(CPU_BOOT_STATUS_IN_WFE);
-		hl_dev_reset();	/* should not return */
+		module_lkd_run_noret_reset();
 		break;
 	case KMD_MSG_FIT_RDY:
-		preboot_security_cleanup_before_run();
-		hl_ts_shutdown();
-		/* We do not know if the FW we are loading supports hard rst */
-		cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_FW_HARD_RST_EN,
-									false);
-		fit_run_image();
+		module_lkd_run_noret_exec();
 		/* no break */
 	default:
 		break;
